feat(mobile): Tab Layout, Profile Page Design, Quiz Card Component

YavizGuldalf opened the pull request:
This PR includes a plethora of changes related to the mobile app. You can find a full list of changes below:

* The old Navbar component has been removed from pages in favor of a tab layout for an overall better look & feel.
* The profile pages have been implemented with mock data, including:
    * A new UserCard component used throughout the profile pages
    * View own & other's profile - See Created/Solved Quizzes, Created Forum Posts/Comments
    * Followers/Following
    * Liked/Bookmarked Quizzes/Posts/Comments
    * Edit Profile page with option to take a picture or choose from the gallery
* The quiz pages have been moved to the new tab layout
* The quiz card component found in the old quiz feed has been extracted to app/components/quizCard.tsx
* The home page has been moved to the new tab layout.
* Links found throughout the app have been adjusted to work with the new layout.

Fixes the Issues #466 and #467.

aoengin commented on mobile/bulingo/app/components/quizCard.tsx:
```diff
@@ -0,0 +1,124 @@
+    right: 20, 
+  },
+  icon: {
+    width: 30,
+    height: 30,
```
Should be relative to the screen resolution

aoengin commented on mobile/bulingo/app/components/quizCard.tsx:
```diff
@@ -0,0 +1,124 @@
+  likeButton: {
+    position: 'absolute',
+    bottom: 20,
+    left: -150,
```
Should be relative to the screen resolution

aoengin commented on mobile/bulingo/app/components/quizCard.tsx:
```diff
@@ -0,0 +1,124 @@
+  bookmarkButton: {
+    position: 'absolute',
+    bottom: 20,
+    right: 20, 
```
Should be relative to the screen resolution

aoengin commented on mobile/bulingo/app/(tabs)/index.tsx:
```diff
@@ -1,17 +1,127 @@
+    margin: 10,
+  },
+  profilePicture: {
+    width: 192,
+    height: 192,
```
Should be relative to the screen resolution

aoengin commented:
As the reviewer of these two issues, I've added some comments on the code.

- Since the quiz card has been extracted and is ready for use, you can replace the placeholders in the profile with actual quiz cards.

<img width="349" alt="Screenshot 2024-11-04 at 01 16 47" src="https://github.com/user-attachments/assets/362c8e62-4a72-46af-8684-2e2db85ef3c6">

- The Following and Followers sections need to be redesigned to include a separator between users. Adding a shadow effect to the "user cards" would also help align them with the rest of the application's design and front-end.

<img width="349" alt="Screenshot 2024-11-04 at 01 16 15" src="https://github.com/user-attachments/assets/d6d8f583-6f3c-4a57-8458-82967b43806e">

- The About Me section expands indefinitely when a user types, making the button and other sections unreachable after a certain point.

<img width="349" alt="Screenshot 2024-11-04 at 01 26 12" src="https://github.com/user-attachments/assets/2c2da8e9-25a1-497c-b1d4-25fdf46c8c85">

- The User Level on the profile is clickable. If this is unintentional, it should be fixed to be non-pressable.

- The text and background colors of the Edit Profile section should be adjusted for better visibility.

- I'm not sure if this is intentional, but when I navigate to the "Other Profile" view, the title appears as shown in the screenshot below. It should be corrected if unintended.

<img width="349" alt="Screenshot 2024-11-04 at 01 36 06" src="https://github.com/user-attachments/assets/e2173534-9a24-4c3e-810b-a09cd67848bb">

YavizGuldalf commented:
@aoengin, I have made the changes you requested. Specifically, 
* Updated the quiz card component to not use absolute positioning
* Replaced placeholders with QuizCard components in profile and other user profile pages
* Updated the design of the edit profile page
* Updated the user card with shadows and background color
* Made 'User Level' unpressable in the profile pages.
* The header in other user profile is now the name of the user instead of users/[username]

You can find screenshots of the changed pages below: 

<img width=300 src="https://github.com/user-attachments/assets/4b2897ec-6d35-4c3f-ac0d-55a6b256985d"/> 
<img width=300 src="https://github.com/user-attachments/assets/ae1a424f-521a-45df-98c1-f3d036e9d539"/>   
<img width=300 src="https://github.com/user-attachments/assets/647d080b-8fc6-4f36-825e-ceb3f01ac10b"/>
<img width=300 src="https://github.com/user-attachments/assets/ed22b785-7ac1-4f2b-b5fc-66e744f0cd53"/>   
<img width=300 src="https://github.com/user-attachments/assets/f2fdb028-e65d-4051-a726-89afa630da77"/>
<img width=300 src="https://github.com/user-attachments/assets/13fa2d54-d2bd-41d6-ab54-759c2c10b60a"/>


> The About Me section expands indefinitely when a user types, making the button and other sections unreachable after a certain point.
The edit profile page having a resizable 'About Me' input box was intentional. That page is scrollable so every item can be viewed, as you can see in the screenshots.

Can you please review these changes?

aoengin commented:
Thanks for the changes. 



<img width="274" alt="Screenshot 2024-11-09 at 11 52 54" src="https://github.com/user-attachments/assets/db6f25df-f04a-49ec-a4ad-70caf41b5cb9">

You can change the text color to white.

> That page is scrollable so every item can be viewed, as you can see in the screenshots.

It's unnecessary and does not appear professional. Please try entering an example description in the quiz creation settings.

YavizGuldalf commented:
> Thanks for the changes.
> 
> <img alt="Screenshot 2024-11-09 at 11 52 54" width="274" src="https://private-user-images.githubusercontent.com/85455415/384590963-db6f25df-f04a-49ec-a4ad-70caf41b5cb9.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzExNTYwMjUsIm5iZiI6MTczMTE1NTcyNSwicGF0aCI6Ii84NTQ1NTQxNS8zODQ1OTA5NjMtZGI2ZjI1ZGYtZjA0YS00OWVjLWE0YWQtNzBjYWY0MWI1Y2I5LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDExMDklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMTA5VDEyMzUyNVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTkzN2ZiNzdmNWQzZWNhN2Y2ZmFjZmRkNTg5YmMyZjliZGFmNjQyNmJmMzZjZDE0NmEwYTE5MjA5Nzg3MTE3NTEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.mlY0zxwic4fUgs9mEs8OJnze9wyBIDYqESprYk9LifI">
> You can change the text color to white.
> 
> > That page is scrollable so every item can be viewed, as you can see in the screenshots.
> 
> It's unnecessary and does not appear professional. Please try entering an example description in the quiz creation settings.

Okay, the 'Edit Profile' button's style has been changed to be the same as the ones in the edit profile page. Also, the 'About Me' section of the edit profile page now has a fixed width and a character limit (120 chars). I have also added a 40 char limit to the name field and fixed its height

aoengin commented:
I guess it's okay to have a character limit, but why didn’t you make it the same as the description part in the quiz settings? I already suggested that.

> Please try entering an example description in the quiz creation settings.

aoengin reviewed (approved):
The fixed character limit may be increased or removed in the future. Other than that, LGTM.

YavizGuldalf commented:
> I guess it's okay to have a character limit, but why didn’t you make it the same as the description part in the quiz settings? I already suggested that.

The main reason why I put a character limit was so the profile page wouldn't look weird if the user tried to enter a really really long "About Me". Without a limit they could write so many characters that it would be unreasonable to display it in their profile. Of course, 120 may not be the appropriate number for the 'About Me' section (it can be lower or higher, it's up for debate), but I think having a character limit on almost all of our inputs is good practice.

> Please try entering an example description in the quiz creation settings.
I don't think I quite understand what you mean by this. The "Description" input part in the quizCreationSettings page looks like the following:

```typescript
<TextInput
  style={[styles.input, styles.descriptionInput]}
  placeholder="Enter quiz description"
  value={quizDescription}
  onChangeText={setQuizDescription}
  multiline={true}
 />

...

styles = StyleSheet.create({
  input: {
    backgroundColor: '#E8E8E8',
    borderRadius: 10,
    padding: 10,
    fontSize: 16,
    marginBottom: 20,
  },
  descriptionInput: {
    height: 100, 
    textAlignVertical: 'top',
  },
...
```

Do you want me to use the same style props with `multiline` set to True?
